app/src/main/java/edu/umd/cmsc436/mstestsuite/MainActivity.java

```diff
@@ -245,6 +248,38 @@ public void onClick(View v) {
     }
 
     @Override
+    public void showHistoryDialog(final String user) {
+        final Dialog dialog = new AppCompatDialog(this);
+        final String [] app_array = getResources().getStringArray(R.array.display_names);
+
+        View root = dialog.getLayoutInflater().inflate(R.layout.history_chooser, null, false);
+
+        TextView title = new TextView(getContext());
+        title.setText("Show score history for: ");
+        title.setTextSize(TypedValue.COMPLEX_UNIT_SP, 25);
+
+        ListView lv = (ListView) root.findViewById(R.id.app_history_listview);
+        lv.addHeaderView(title);
+
+        lv.setAdapter(new ArrayAdapter<>(this, R.layout.plain_list_item, R.id.listview_item_textview, app_array));
+        lv.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                dialog.dismiss();
+                String app_name = "edu.umd.cmsc436." + app_array[position-1];
+
+                Intent i = new Intent(app_name + ".action.HISTORY");
+                i.putExtra("user", user);
+                i.addCategory(Intent.CATEGORY_DEFAULT);
+                startActivity(i);
```

Comment:
This can be wrapped in a try-catch block looking for an ActivityNotFoundException, which if it is caught, can call showToast() to explain the error.  (I know I said it was fine earlier, but while changes are being made it'd be convenient, thanks!)